hazelcast/src/main/java/com/hazelcast/jet/impl/jobupload/JobMetaDataParameterObject.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.jobupload;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * The wrapper for all parameters to run an uploaded jar as Jet job
+ */
+public class JobMetaDataParameterObject {
```

Comment:
Yes they are similar but they are at different layers of the applicaiton. 
The parameter passed to submitJobFromJar is a client binary protocol message. 
JobMetaDataParameterObject is at the business logic. So they should be separate

Reply:
I'd just change the name a bit, e.g. `JobUploadMetadata`?

Also, I think it can be safely changed to `record`-style class, so every field final.
+ add please toString for easier debugging in the future